custom/icds_reports/static/js/angular-services/locations.service.js

```diff
@@ -122,19 +131,191 @@ window.angular.module('icdsApp').factory('locationsService', ['$http', '$locatio
             return (
                 level === 0 ||  // first level to select should be visible
                 // or previous location should be set and not equal to "all".
-                (selectedLocations[level - 1] && selectedLocations[level - 1] !== 'all' && selectedLocations[level - 1].location_id !== 'all')
+                    (selectedLocations[level - 1] && selectedLocations[level - 1] !== ALL_OPTION.location_id && selectedLocations[level - 1].location_id !== ALL_OPTION.location_id)
             );
         },
         getLocationsForLevel: function (level, selectedLocations, locationsCache) {
             if (level === 0) {
                 return locationsCache.root;
             } else {
                 var selectedLocation = selectedLocations[level - 1];
-                if (!selectedLocation || selectedLocation.location_id === 'all') {
+                if (!selectedLocation || selectedLocation.location_id === ALL_OPTION.location_id) {
                     return [];
                 }
                 return locationsCache[selectedLocation.location_id];
             }
         },
+
+        initHierarchy : function(locationHierarchy) {
```

Comment:
how much of this is new code versus just copied in? it looks like most of this comes from code that was deleted in the next commit - wondering how closely it should be reviewed

Reply:
This is pretty much entirely copied from `download.directive.js`. Most changes are just adding extra parameters that were not explicitly passed in. I think generally, it should not have to be reviewed too closely.

Reply:
:+1: ok that was my guess too (and therefore i won't review closely / punish you for someone else's code0